Adyen/Core/Core Protocols/Details.swift

```diff
@@ -16,5 +16,25 @@ public protocol PaymentMethodDetails: Details {
     var checkoutAttemptId: String? { get set }
 }
 
+public extension PaymentMethodDetails {
+
+    @_spi(AdyenInternal)
+    var checkoutAttemptId: String? {
```

Comment:
Is there a way to check if the library is built with `BUILD_LIBRARY_FOR_DISTRIBUTION` and only conditionally add the default conformance to make the compiler happy?
This way we would not need to provide this implementation and use objc_getAssociatedObject.